File: sdkwork-backend-sdk-flutter/lib/src/api/generation_content.dart
import '../http/client.dart';
import '../models.dart';

class GenerationContentApi {
  final HttpClient _client;
  
  GenerationContentApi(this._client);

  /// Update AI generated content
  Future<PlusApiResultPlusAiGenerationContentVO?> update(PlusAiGenerationContentForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/generation/content'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusAiGenerationContentVO ? response : null;
  }

  /// Create new AI generated content
  Future<PlusApiResultPlusAiGenerationContentVO?> create(PlusAiGenerationContentForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/generation/content'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusAiGenerationContentVO ? response : null;
  }

  /// Get AI generated content by page
  Future<PlusApiResultPagePlusAiGenerationContentVO?> listByPage(QueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/generation/content/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusAiGenerationContentVO ? response : null;
  }

  /// Get all AI generated content
  Future<PlusApiResultListPlusAiGenerationContentVO?> listAllEntities(QueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/generation/content/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListPlusAiGenerationContentVO ? response : null;
  }

  /// Get AI generated content by ID
  Future<PlusApiResultPlusAiGenerationContentVO?> getById(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/generation/content/${id}'));
    return response is PlusApiResultPlusAiGenerationContentVO ? response : null;
  }

  /// Delete AI generated content
  Future<PlusApiResultBoolean?> delete(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/generation/content/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }
}

File: sdkwork-backend-sdk-typescript/src/api/generation-content.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusAiGenerationContentForm, PlusApiResultBoolean, PlusApiResultListPlusAiGenerationContentVO, PlusApiResultPagePlusAiGenerationContentVO, PlusApiResultPlusAiGenerationContentVO, QueryListForm } from '../types';


export class GenerationContentApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Update AI generated content */
  async update(body: PlusAiGenerationContentForm): Promise<PlusApiResultPlusAiGenerationContentVO> {
    return this.client.put<PlusApiResultPlusAiGenerationContentVO>(backendApiPath(`/generation/content`), body);
  }

/** Create new AI generated content */
  async create(body: PlusAiGenerationContentForm): Promise<PlusApiResultPlusAiGenerationContentVO> {
    return this.client.post<PlusApiResultPlusAiGenerationContentVO>(backendApiPath(`/generation/content`), body);
  }

/** Get AI generated content by page */
  async listByPage(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusAiGenerationContentVO> {
    return this.client.post<PlusApiResultPagePlusAiGenerationContentVO>(backendApiPath(`/generation/content/list`), body, params);
  }

/** Get all AI generated content */
  async listAllEntities(body?: QueryListForm): Promise<PlusApiResultListPlusAiGenerationContentVO> {
    return this.client.post<PlusApiResultListPlusAiGenerationContentVO>(backendApiPath(`/generation/content/list/all`), body);
  }

/** Get AI generated content by ID */
  async getById(id: string | number): Promise<PlusApiResultPlusAiGenerationContentVO> {
    return this.client.get<PlusApiResultPlusAiGenerationContentVO>(backendApiPath(`/generation/content/${id}`));
  }

/** Delete AI generated content */
  async delete(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/generation/content/${id}`));
  }
}

export function createGenerationContentApi(client: HttpClient): GenerationContentApi {
  return new GenerationContentApi(client);
}
